enhancements/insights/insights_gateway.md

```diff
@@ -0,0 +1,185 @@
+(example in [cost management documentation](https://access.redhat.com/documentation
+/en-us/openshift_container_platform/4.5/html/getting_started_with_cost_management/
+assembly_adding_sources_cost#configuring_cost_mgmt-operator)). This leads to:
+
+- increased complexity for the user to set up this components
+- the need for each component to implement their own way to send the data 
+- number of places the credentials to cloud.redhat.com are stored
+- no single place to audit and opt-out form the payloads being sent to c.r.c.
+
+Given insights-operator is already part of the OCP cluster and has proper credentials
+configured already as part of the installation, we're proposing for it to expose
+a proxy for other components to be able to send the payloads on their behalf.
+
+### Goals
+
+1. Streamline the process for additional components to send data from the OCP cluster.
+2. No need for each component to keep their own credentials.
+3. Basic centralized overview about payloads being sent to c.r.c.
+4. Ability to opt-out form all the payloads being sent to c.r.c., leveraging the machanism
+that's already in place in insights-operator.
+
+### Non-Goals
+
+1. Build and maintain a separate component to acts as a gateway: this can be
+   subject of additional proposal.
+2. Provide additional guaranties about the content of the payload beyond basic logging
+   (e.g. time, source, size). The information about the data being collected and it's
+   purpose needs to be documented by the operator collecting the data.
+3. Granular control over allowed/denied payload types: for the sake of this proposal, it's 
+   all or nothing.
+
+## Proposal
+
+1. insights-operator will expose a service available within the OCP cluster
+   that would provide an endpoint to proxy the payloads to to c.r.c. using
+   its pull-secret credentials. The endpoint would process POST requests
+   at the following URL:
+   
+   ```
+   https://insights-operator.openshift-insights.svc/rhinsights/v1/upload
+   ```
+
+2. insights-operator will define a `ClusterRole`` granting access to this API via 'nonResourceURLs':
+
+```
+apiVersion: rbac.authorization.k8s.io/v1
+kind: ClusterRole
+metadata:
+  name: rhinsights-upload
+rules:
+- nonResourceURLs:
+  - /rhinsights/v1/upload
+  verbs:
+  - post
+```
+3. for the third-party component to use the endpoint, it will bind a service account to the cluster role:
+
+```
+---
+apiVersion: rbac.authorization.k8s.io/v1
+kind: ClusterRoleBinding
+metadata:
+  name: olm-operator
+  namespace: olm-operator
+roleRef:
+  apiGroup: rbac.authorization.k8s.io
+  kind: ClusterRole
+  name: rhinsights-upload
+subjects:
+  - kind: ServiceAccount
+    name: olm-operator
+    namespace: olm-operator
+```
+4. before forwarding the request, the io will:
+  - log the event of forwarding the request (service account info, content_type, payload size…)
+  - add additional headers to the request about it's origin (service account etc… - TBD)
```

Comment:
@jhjaggars what kind of headers you think would be useful here for the ingress service?